### ISSUE TITLE
after OK postfixadmin-cli population of alias table, UI alias list view query fails : " Invalid query: near "AS": syntax error"

### ISSUE THREAD

**Issue by pgnd** [postfixadmin/postfixadmin#387]
i'm migrating a bunch of aliases to a new pfa-managed server, from another server.

i've

	cd src/postfixadmin/
	git log -n1
      1 commit f7708520b36b07598db641bd076eee46fbdbf7af (HEAD, tag: postfixadmin-3.2.4)
      2 Author: David Goodwin <david@codepoets.co.uk>
      3 Date:   Fri Apr 17 09:15:04 2020 +0100
      4
      5     postfixadmin 3.2.4 changelog
	php -v
		PHP 7.4.10 (cli) (built: Sep  1 2020 13:58:08) ( NTS )
		Copyright (c) The PHP Group
		Zend Engine v3.4.0, Copyright (c) Zend Technologies
			with Zend OPcache v7.4.10, Copyright (c), by Zend Technologies
	sqlite3 -version
		3.33.0 2020-08-14 13:23:32 fca8dc8b578f215a969cd899336378966156154710873e68b3d9ac5881b0alt1

i've extracted existing alias maps into a source text file.

it's in simple text format, one map per line,
```
	aliasaddress@domain1.com  gotoaddress@domain2.com
```

i've imported all the aliases with a simple 'postfixadm-cli' shell script
```
	THIS_ALIAS_FILE="~/virtual_alias_maps.ORIG"
	IFS=' '
	while read -r line
	do
	 if [[ -n "$line" && "$line" != [[:blank:]#]* ]]
	 then
	  LINE_DATA=()
	  read -ra LINE_DATA <<< "$line"
	  THIS_ALIAS="${LINE_DATA[0]}"
	  THIS_TARGET="${LINE_DATA[1]}"
	  postfixadmin-cli alias add ${THIS_ALIAS} --goto ${THIS_TARGET} --active 1
	 fi
	done < ${THIS_ALIAS_FILE}
```

it completes without error.

checking sqlite's ALIAS table, all looks good.  both the original/default aliases
```
	address                         |goto                            |domain                     |created            |modified           |active|
	--------------------------------|--------------------------------|---------------------------|-------------------|-------------------|------|
	admin@mx.example.net            |admin@mx.example.net            |mx.example.net             |2020-09-14 19:11:56|2020-09-14 19:11:56|     1|
	sender@mx.example.net           |sender@mx.example.net           |mx.example.net             |2020-09-14 19:11:57|2020-09-14 19:11:57|     1|
	postmaster@example.com          |postmaster@example.com          |example.com                |2020-09-14 19:11:58|2020-09-14 19:11:58|     1|
	user1@example.com               |user1@example.com               |example.com                |2020-09-14 19:11:58|2020-09-14 19:11:58|     1|
	user2@example.com               |user2@example.com               |example.com                |2020-09-14 19:11:58|2020-09-14 19:11:58|     1|
	providers@example.com           |providers@example.com           |example.com                |2020-09-14 19:11:59|2020-09-14 19:11:59|     1|
	postmaster@example2.com         |postmaster@example2.com         |example2.com               |2020-09-14 19:11:59|2020-09-14 19:11:59|     1|
	user4@example2.com              |user4@example2.com              |example2.com               |2020-09-14 19:11:59|2020-09-14 19:11:59|     1|
	user3@example2.com              |user3@example2.com              |example2.com               |2020-09-14 19:11:59|2020-09-14 19:11:59|     1|
	postmaster@example3.com         |postmaster@example3.com         |example3.com               |2020-09-14 19:12:00|2020-09-14 19:12:00|     1|
	user1@example3.com              |user1@example3.com              |example3.com               |2020-09-14 19:12:00|2020-09-14 19:12:00|     1|
```

as well as the script-added maps,
```
	test@example.com                |user1@example.com               |example.com                |2020-09-14 19:18:56|2020-09-14 19:18:56|     1|
	root@mx.example.net             |admin@mx.example.net            |mx.example.net             |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	postmaster@mx.example.net       |admin@mx.example.net            |mx.example.net             |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	admin2@mx.example.net           |admin@mx.example.net            |mx.example.net             |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	testalias@example3.com          |user1@example.com               |example3.com               |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	user1.mobile@example.com        |user1@example.com               |example.com                |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	a.a@example.com                 |user1@example.com               |example.com                |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	b-b@example.com                 |user1@example.com               |example.com                |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	c.c@example.com                 |user1@example.com               |example.com                |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	user3.a.a@example.com           |user1@example.com               |example.com                |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	user3.b@example.com             |user1@example.com               |example.com                |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	user3.c@example.com             |user1@example.com               |example.com                |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	spamtest@example.com            |user1@example.com               |example.com                |2020-09-14 19:22:16|2020-09-14 19:22:16|     1|
	...
```

after this^ exec, nav to the ALIAS list tab in the UI, @

	https://postfixadmin.example.net/list-virtual.php?domain=example.com&limit=0

returns in browser
```
	DEBUG INFORMATION:
	Invalid query: near "AS": syntax error

	Check your error_log for the failed query.

	Please check the documentation and website for more information.

	Postfix Admin
	Forums 
```

and in logs
```
	[14-Sep-2020 12:40:09 America/Los_Angeles] PHP Notice:  Undefined offset: 1 in /usr/local/src/postfixadmin/functions.inc.php on line 554
	[14-Sep-2020 12:40:09 America/Los_Angeles] Invalid query: near "AS": syntax error
	[14-Sep-2020 12:40:09 America/Los_Angeles] caused by query:
				WITH idx AS (SELECT *  FROM alias  LEFT JOIN (  SELECT 1 as __is_mailbox, username as __mailbox_username  FROM mailbox WHERE username IS NOT NULL  AND  domain IN ('example.com')  ) AS __mailbox ON __mailbox_username = address
	WHERE 1=1   AND  domain IN ('example2.com','example3.com','example4.com','example5.com','example6.com','mx.example.net','example7.com','example8.com','example.com','example9.com')   AND    ( domain='example.com' AND __mailbox_username IS NULL )  ORDER BY address)
					SELECT  AS label, (SELECT (COUNT(*) - 1) FROM idx t1 WHERE t1. <= t2. ) AS r
					FROM idx t2
					WHERE (r % 10) IN (0,9) OR r = 25
```

**Comment by pgnd** [comment]
the cuplrit appears to be
```
// Page Size
// Set the number of entries that you would like to see
// in one page.
$CONF['page_size'] = '10';
```

if #n aliases are added, redirecting to the same target, so that -- including the default alias<->goto map -- there are a total of #(n+1) aliases for given target,

if the value of
```
	$CONF['page_size'] < #n
```

, the error as above -- in display and in query log -- fires.

as long as
```
	$CONF['page_size'] >= #n
```

the page displays normally; no error in browser or app logs.

this (mis)behavior is fully reproducible on 3 separate installs; admittedly similarly configured.

checking src,
```
postfixadmin/public//list-virtual.php
```

indeed has a slew of 'page_size' conditionals ... which/why is the question.

**Comment by DavidGoodwin** [comment]
https://github.com/postfixadmin/postfixadmin/blob/432f02472f8f466f172db2b46b043ff593363883/functions.inc.php#L556 is perhaps is to blame.

Which looks like #87 and #161 are still an issue ?

**Comment by jrd** [comment]
Same problem here.

**Comment by DavidGoodwin** [comment]
I can't replicate this problem with 'master' ..... I assume you're using Postfixadmin 3.2.x ?

**Comment by pgnd** [comment]
currently, here:
```
git branch -a
      1 * (HEAD detached at postfixadmin-3.2.4)
      2   master
      3   remotes/origin/HEAD -> origin/master
      4   remotes/origin/feature-php-crypt-prefix
      5   remotes/origin/gh-pages
      6   remotes/origin/master
      7   remotes/origin/postfixadmin-2.3
      8   remotes/origin/postfixadmin_3.2
git log -n1
      1 commit f7708520b36b07598db641bd076eee46fbdbf7af (HEAD, tag: postfixadmin-3.2.4)
      2 Author: David Goodwin <david@codepoets.co.uk>
      3 Date:   Fri Apr 17 09:15:04 2020 +0100
      4
      5     postfixadmin 3.2.4 changelog
```

of related, minor note, @

	https://postfixadmin.example.com/main.php

the "Check for update" link URI is
```
http://postfixadmin.sourceforge.net/update-check.php?version=3.2
```
( btw, might be nice to have an http*s*// link here ... )

& returns
```
Upgrade available - the latest version is 3.2.4
```

which I assume is simply a red-herring ...
